cpp/src/arrow/io/compressed-test.cc

```diff
@@ -73,7 +73,7 @@ std::shared_ptr<Buffer> CompressDataOneShot(Codec* codec,
   ABORT_NOT_OK(codec->Compress(data.size(), data.data(), max_compressed_len,
                                compressed->mutable_data(), &compressed_len));
   ABORT_NOT_OK(compressed->Resize(compressed_len));
-  return compressed;
+  return std::move(compressed);
```

Comment:
I'm confused that this is needed. I would normally think that we should not `std::move` here as return-value-optimization should kick in.